pallets/offchain-worker/src/utils.rs

```diff
@@ -0,0 +1,84 @@
+//! Based on https://github.com/debris/base58/blob/master/src/lib.rs
+//! works only up to 128 bytes
+use sp_std::prelude::*;
+
+const ALPHABET: &'static [u8] = b"123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+
+/// A trait for converting a value to base58 encoded string.
+pub trait ToBase58 {
+	/// Converts a value of `self` to a base58 value, returning the owned string.
+	fn to_base58(&self) -> Vec<u8>;
+}
+
+impl ToBase58 for [u8] {
+	fn to_base58(&self) -> Vec<u8> {
+		let zcount = self.iter().take_while(|x| **x == 0).count();
+        let size = (self.len() - zcount) * 138 / 100 + 1;
+		let mut buffer = vec![0u8; size];
+
+		let mut i = zcount;
+		let mut high = size - 1;
+
+		while i < self.len() {
+			let mut carry = self[i] as u32;
+			let mut j = size - 1;
+
+			while j > high || carry != 0 {
+				carry += 256 * buffer[j] as u32;
+				buffer[j] = (carry % 58) as u8;
+				carry /= 58;
+
+				// in original trezor implementation it was underflowing
+				if j  > 0 {
+					j -= 1;
+				}
+			}
+
+			i += 1;
+			high = j;
+		}
+
+		let mut j = buffer.iter().take_while(|x| **x == 0).count();
+
+		let mut result = Vec::new();
+		for _ in 0..zcount {
+			result.push(b'1');
+		}
+
+		while j < size {
+			result.push(ALPHABET[buffer[j] as usize]);
+			j += 1;
+		}
+
+        result
```

Comment:
Identation seems to be off.

Reply:
Seems to be a problem of the review preview but not the code